core/src/main/scala/org/apache/spark/status/protobuf/ExecutorSummaryWrapperSerializer.scala

```diff
@@ -0,0 +1,181 @@
+      .setTotalGcTime(input.totalGCTime)
+      .setTotalInputBytes(input.totalInputBytes)
+      .setTotalShuffleRead(input.totalShuffleRead)
+      .setTotalShuffleWrite(input.totalShuffleWrite)
+      .setIsBlacklisted(input.isBlacklisted)
+      .setMaxMemory(input.maxMemory)
+      .setAddTime(input.addTime.getTime)
+
+    input.removeTime.foreach {
+      date => builder.setRemoveTime(date.getTime)
+    }
+    input.removeReason.foreach(builder.setRemoveReason)
+    input.executorLogs.foreach { case (k, v) =>
+      builder.putExecutorLogs(k, v)
+    }
+    input.memoryMetrics.foreach { metrics =>
+      builder.setMemoryMetrics(serializeMemoryMetrics(metrics))
+    }
+    input.blacklistedInStages.foreach(builder.addBlacklistedInStages)
+    input.peakMemoryMetrics.foreach { metrics =>
+      builder.setPeakMemoryMetrics(ExecutorMetricsSerializer.serialize(metrics))
+    }
+    input.attributes.foreach { case (k, v) =>
+      builder.putAttributes(k, v)
+    }
+    input.resources.foreach { case (k, v) =>
+      builder.putResources(k, serializeResourceInformation(v))
+    }
+
+    builder.setResourceProfileId(input.resourceProfileId)
+    builder.setIsExcluded(input.isExcluded)
+
+    input.excludedInStages.foreach(builder.addExcludedInStages)
+
+    builder.build()
+  }
+
+  private def deserializeExecutorSummary(
+      binary: StoreTypes.ExecutorSummary): ExecutorSummary = {
+    val peakMemoryMetrics =
+      getOptional(binary.hasPeakMemoryMetrics,
+        () => ExecutorMetricsSerializer.deserialize(binary.getPeakMemoryMetrics))
+    val removeTime = getOptional(binary.hasRemoveTime, () => new Date(binary.getRemoveTime))
+    val removeReason = getOptional(binary.hasRemoveReason, () => binary.getRemoveReason)
+    val memoryMetrics =
+      getOptional(binary.hasMemoryMetrics,
+        () => deserializeMemoryMetrics(binary.getMemoryMetrics))
+    new ExecutorSummary(
+      id = binary.getId,
+      hostPort = binary.getHostPort,
+      isActive = binary.getIsActive,
+      rddBlocks = binary.getRddBlocks,
+      memoryUsed = binary.getMemoryUsed,
+      diskUsed = binary.getDiskUsed,
+      totalCores = binary.getTotalCores,
+      maxTasks = binary.getMaxTasks,
+      activeTasks = binary.getActiveTasks,
+      failedTasks = binary.getFailedTasks,
+      completedTasks = binary.getCompletedTasks,
+      totalTasks = binary.getTotalTasks,
+      totalDuration = binary.getTotalDuration,
+      totalGCTime = binary.getTotalGcTime,
+      totalInputBytes = binary.getTotalInputBytes,
+      totalShuffleRead = binary.getTotalShuffleRead,
+      totalShuffleWrite = binary.getTotalShuffleWrite,
+      isBlacklisted = binary.getIsBlacklisted,
+      maxMemory = binary.getMaxMemory,
+      addTime = new Date(binary.getAddTime),
+      removeTime = removeTime,
+      removeReason = removeReason,
+      executorLogs = binary.getExecutorLogsMap.asScala.toMap,
+      memoryMetrics = memoryMetrics,
+      // TODO: fix toInt
```

Comment:
@techaddict There are some todos here. Will you fix them?

Reply:
done, made stageIds int64

Reply:
@gengliangwang  done, made stageIds int64